### DISCUSSION TITLE
How to find which modules the module is imported by?

### DISCUSSION THREAD

**Discussion by ahabhgk** [webpack/webpack#13041] [Q&A]
```js
// page1.js
import img from './img.png'
console.log('page1: ', img)
```

```js
// page2.js
import img from './img.png'
console.log('page2: ', img)
```

In webpack4, I want to find which modules import the img, I thought modules.reasons maybe possible in this issue, but when I writing a plugin, in normalModuleLoader hook, the img module.reasons have two modules, both are page1 module.

How do I get page1 module and page2 module in normalModuleLoader hook?

And how to do this in webpack5?

**Comment by alexander-akait** [comment]
`issuer`?

**Reply by alexander-akait** [reply]
Also you should avoid using `import something from './file.non-js-ext'`, please prefer to use `const img = new URL('./image.png', import.meta.url);`

**Reply by ahabhgk** [reply]
> `issuer`?

`module.reasons`

Forgot to attach the issue's link: [module.issuer vs. module.reasons](https://github.com/webpack/webpack/issues/10647)

**Comment by sokra** [comment]
In webpack 5: `moduleGraph.getIncomingConnections`

**Reply by ahabhgk** [reply]
Thanks, it works in webpack5. by the way, what's the difference between resolvedOriginModule and originModule?
